src/parents.js

```diff
@@ -26,24 +26,41 @@ export function setAll (node, options ) {
  * Set the `parent` property on a node.
  * By default it will skip nodes that already have a `parent` property, but you can set force = true to prevent that.
  * @param {object} node
- * @param {object} parent
+ * @param {object | null} parentPath
  * @param {object} [options]
  * @param {boolean} [options.force] Allow overwriting
  */
-export function set (node, parent, { force } = {}) {
+export function set (node, parentPath, { force } = {}) {
 	if (!force && parentMap.has(node)) {
 		// We assume that if the node already has a parent, its subtree will also have parents
 		return false;
 	}
-	parentMap.set(node, parent);
+
+	if (parentPath === null) {
+		// Node is the AST root
+		parentMap.set(node, { node: null });
+		return;
+	}
```

Comment:
One thing I'm unsure of: how do we want to handle root nodes? Do we want to include them as a `parentPath` with `node` set to `null` and `property` and `index` `undefined` (like above), or do we want to have it as null in the map, i.e. 
```js
parentMap.set(node, null);
```

Reply:
Since there is no parent, there is no parent path, so I vote `null` for the whole thing.

Reply:
Great, we'll go with that